src/plots/cartesian/position_defaults.js

```diff
@@ -36,16 +40,31 @@ module.exports = function handlePositionDefaults(containerIn, containerOut, coer
         }
     }, 'anchor');
 
-    if(anchor === 'free') coerce('position', dfltPosition);
-
-    Lib.coerce(containerIn, containerOut, {
+    var side = Lib.coerce(containerIn, containerOut, {
         side: {
             valType: 'enumerated',
             values: letter === 'x' ? ['bottom', 'top'] : ['left', 'right'],
             dflt: dfltSide
         }
     }, 'side');
 
+    if(anchor === 'free') {
+        if(shift !== false && shift !== undefined) {
```

Comment:
After considering my comment here: https://github.com/plotly/plotly.js/pull/6334/files#r1046315695,
you may move this block right after `shift` coerce.